fmc/src/flow/pcr.rs

```diff
@@ -23,49 +23,87 @@ Note:
 
-/// Extend current PCR
-///
-/// # Arguments
-///
-/// * `env` - FMC Environment
-pub fn extend_current_pcr(env: &mut FmcEnv, hand_off: &HandOff) -> CaliptraResult<()> {
-    // Clear current PCR before extending it.
-    if env.soc_ifc.reset_reason() == caliptra_drivers::ResetReason::UpdateReset {
-        env.pcr_bank.erase_pcr(RT_FW_CURRENT_PCR)?;
-    }
-    extend_pcr_common(env, hand_off, RT_FW_CURRENT_PCR)
-}
-
-/// Extend journey PCR
-///
-/// # Arguments
-///
-/// * `env` - FMC Environment
-pub fn extend_journey_pcr(env: &mut FmcEnv, hand_off: &HandOff) -> CaliptraResult<()> {
-    extend_pcr_common(env, hand_off, RT_FW_JOURNEY_PCR)
-}
-
-/// Extend common data into PCR
+/// Extend common data into the current and journey PCRs
 ///
 /// # Arguments
 ///
 /// * `env` - FMC Environment
 /// * `pcr_id` - PCR slot to extend the data into
-fn extend_pcr_common(env: &mut FmcEnv, hand_off: &HandOff, pcr_id: PcrId) -> CaliptraResult<()> {
-    // Extend RT TCI (Hash over runtime code)
+///
+/// TODO: Add CFI instrumentation
+pub fn extend_pcr_common(env: &mut FmcEnv, hand_off: &mut HandOff) -> CaliptraResult<()> {
+    // Calculate RT TCI (Hash over runtime code)
     let rt_tci = Tci::rt_tci(env, hand_off);
     let rt_tci: [u8; 48] = okref(&rt_tci)?.into();
-    env.pcr_bank.extend_pcr(pcr_id, &mut env.sha384, &rt_tci)?;
 
-    // Extend FW Image Manifest
+    // Calculate FW Image Manifest digest
     let manifest_digest = Tci::image_manifest_digest(env, hand_off);
     let manifest_digest: [u8; 48] = okref(&manifest_digest)?.into();
+
+    // Clear current PCR before extending it.
+    env.pcr_bank.erase_pcr(RT_FW_CURRENT_PCR)?;
+
+    extend_and_log(env, PcrLogEntryId::RtTci, &rt_tci)?;
+    extend_and_log(env, PcrLogEntryId::FwImageManifest, &manifest_digest)?;
+
+    Ok(())
+}
+
+/// Extend `data` into both the current and journey PCRs, and updates the PCR log.
+/// TODO: Add CFI instrumentation
+fn extend_and_log(env: &mut FmcEnv, entry_id: PcrLogEntryId, data: &[u8]) -> CaliptraResult<()> {
     env.pcr_bank
-        .extend_pcr(pcr_id, &mut env.sha384, &manifest_digest)?;
+        .extend_pcr(RT_FW_CURRENT_PCR, &mut env.sha384, data)?;
+    env.pcr_bank
+        .extend_pcr(RT_FW_JOURNEY_PCR, &mut env.sha384, data)?;
+
+    log_pcr(
+        &mut env.persistent_data.get_mut().pcr_log,
+        &mut env.pcr_bank,
+        entry_id,
+        (1 << RT_FW_CURRENT_PCR as u8) | (1 << RT_FW_JOURNEY_PCR as u8),
+        data,
+    )
+}
+
+// TODO: Add CFI instrumentation
+fn log_pcr(
+    pcr_log: &mut PcrLogArray,
+    pcr_bank: &mut PcrBank,
+    pcr_entry_id: PcrLogEntryId,
+    pcr_ids: u32,
+    data: &[u8],
+) -> CaliptraResult<()> {
+    let Some(dst) = pcr_log.get_mut(pcr_bank.log_index) else {
+        return Err(CaliptraError::FMC_GLOBAL_PCR_LOG_EXHAUSTED);
+    };
+
+    // Create a PCR log entry
+    cprintln!("pcr_entry_id: {:?}", pcr_entry_id as u16);
+    cprintln!("FMC log_index: {:?}", pcr_bank.log_index as u16);
```

Comment:
Can these be removed? The information is in the PCR log anyway.

Reply:
Leftover debug lines, removed.